diff --git a/decoder/gym.go b/decoder/gym.go
index 09ecda6..117a42f 100644
--- a/decoder/gym.go
+++ b/decoder/gym.go
@@ -181,7 +181,7 @@ func (gym *Gym) updateGymFromFort(fortData *pogo.PokemonFortProto, cellId uint64
 		TempEvolutionFinishMs int64 `json:"temp_evolution_finish_ms,omitempty"`
 		Alignment             int   `json:"alignment,omitempty"`
 		Badge                 int   `json:"badge,omitempty"`
-		Background            int   `json:"background,omitempty"`
+		Background            int64 `json:"background,omitempty"`
 	}
 	gym.Id = fortData.FortId
 	gym.Lat = fortData.Latitude  //fmt.Sprintf("%f", fortData.Latitude)
@@ -331,7 +331,7 @@ func (gym *Gym) updateGymFromGymInfoOutProto(gymData *pogo.GymGetInfoOutProto) *
 		TempEvolutionFinishMs int64              `json:"temp_evolution_finish_ms,omitempty"`
 		Alignment             int                `json:"alignment,omitempty"`
 		Badge                 int                `json:"badge,omitempty"`
-		Background            int                `json:"background,omitempty"`
+		Background            int64              `json:"background,omitempty"`
 		DeployedMs            int64              `json:"deployed_ms,omitempty"`
 		DeployedTime          int64              `json:"deployed_time,omitempty"`
 		BattlesWon            int32              `json:"battles_won"`
diff --git a/decoder/pokemon.go b/decoder/pokemon.go
index 9894cdf..898c396 100644
--- a/decoder/pokemon.go
+++ b/decoder/pokemon.go
@@ -6,6 +6,7 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"golbat/util"
 	"strconv"
 	"strings"
 	"sync"
@@ -1108,9 +1109,7 @@ func (pokemon *Pokemon) clearIv(cp bool) {
 func (pokemon *Pokemon) addEncounterPokemon(ctx context.Context, db db.DbDetails, proto *pogo.PokemonProto, username string) {
 	pokemon.Username = null.StringFrom(username)
 	pokemon.Shiny = null.BoolFrom(proto.PokemonDisplay.Shiny)
-	if proto.PokemonDisplay.LocationCard != nil {
-		pokemon.Background = null.IntFrom(int64(proto.PokemonDisplay.LocationCard.LocationCard.Number()))
-	}
+	pokemon.Background = null.IntFrom(util.ExtractBackgroundFromDisplay(proto.PokemonDisplay))
 	pokemon.Cp = null.IntFrom(int64(proto.Cp))
 	pokemon.Move1 = null.IntFrom(int64(proto.Move1))
 	pokemon.Move2 = null.IntFrom(int64(proto.Move2))
diff --git a/decoder/pokestop.go b/decoder/pokestop.go
index 0a88fa3..4a3b4f5 100644
--- a/decoder/pokestop.go
+++ b/decoder/pokestop.go
@@ -609,7 +609,7 @@ func (stop *Pokestop) updatePokestopFromGetPokemonSizeContestEntryOutProto(conte
 		TempEvolutionFinishMs int64   `json:"temp_evolution_finish_ms"`
 		Alignment             int     `json:"alignment"`
 		Badge                 int     `json:"badge"`
-		Background            int     `json:"background"`
+		Background            int64   `json:"background"`
 	}
 	type contestJson struct {
 		TotalEntries   int            `json:"total_entries"`
diff --git a/util/pogo.go b/util/pogo.go
index f29dbc2..9c09f0c 100644
--- a/util/pogo.go
+++ b/util/pogo.go
@@ -27,9 +27,9 @@ var IncidentTypeToName = map[int8]string{
 	9: "showcase",
 }
 
-func ExtractBackgroundFromDisplay(display *pogo.PokemonDisplayProto) int {
+func ExtractBackgroundFromDisplay(display *pogo.PokemonDisplayProto) int64 {
 	if display.LocationCard == nil {
 		return 0
 	}
-	return int(display.LocationCard.LocationCard)
+	return int64(display.LocationCard.LocationCard)
 }
